cmd/infracost/comment_bitbucket.go

```diff
@@ -0,0 +1,133 @@
+			ctx.SetContextValue("platform", "bitbucket")
+
+			var err error
+
+			serverURL, _ := cmd.Flags().GetString("bitbucket-server-url")
+			token, _ := cmd.Flags().GetString("bitbucket-token")
+			tag, _ := cmd.Flags().GetString("tag")
+			extra := comment.BitbucketExtra{
+				ServerURL: serverURL,
+				Token:     token,
+				Tag:       tag,
+			}
+
+			commit, _ := cmd.Flags().GetString("commit")
+			prNumber, _ := cmd.Flags().GetInt("pull-request")
+			repo, _ := cmd.Flags().GetString("repo")
+
+			var commentHandler *comment.CommentHandler
+			if prNumber != 0 {
+				ctx.SetContextValue("targetType", "pull-request")
+
+				commentHandler, err = comment.NewBitbucketPRHandler(ctx.Context(), repo, strconv.Itoa(prNumber), extra)
+				if err != nil {
+					return err
+				}
+			} else if commit != "" {
+				ctx.SetContextValue("targetType", "commit")
+
+				commentHandler, err = comment.NewBitbucketCommitHandler(ctx.Context(), repo, commit, extra)
+				if err != nil {
+					return err
+				}
+			} else {
+				ui.PrintUsage(cmd)
+				return fmt.Errorf("either --commit or --pull-request is required")
+			}
+
+			behavior, _ := cmd.Flags().GetString("behavior")
+			if behavior != "" && !contains(validCommentBitbucketBehaviors, behavior) {
+				ui.PrintUsage(cmd)
+				return fmt.Errorf("--behavior only supports %s", strings.Join(validCommentBitbucketBehaviors, ", "))
+			}
+			ctx.SetContextValue("behavior", behavior)
+
+			paths, _ := cmd.Flags().GetStringArray("path")
+
+			body, err := buildCommentBody(cmd, ctx, paths, output.MarkdownOptions{
+				WillUpdate:          prNumber != 0 && behavior == "update",
+				WillReplace:         prNumber != 0 && behavior == "delete-and-new",
+				IncludeFeedbackLink: true,
+				BasicSyntax:         true,
+			})
+			if err != nil {
+				return err
+			}
+
+			dryRun, _ := cmd.Flags().GetBool("dry-run")
+			if !dryRun {
+				err = commentHandler.CommentWithBehavior(ctx.Context(), behavior, string(body))
+				if err != nil {
+					return err
+				}
+
+				pricingClient := apiclient.NewPricingAPIClient(ctx)
+				err = pricingClient.AddEvent("infracost-comment", ctx.EventEnv())
+				if err != nil {
+					log.Errorf("Error reporting event: %s", err)
+				}
+
+				cmd.Println("Comment posted to Bitbucket")
+			} else {
+				cmd.Println(string(body))
+				cmd.Println("Comment not posted to Bitbucket (--dry-run was specified)")
+			}
+
+			return nil
+		},
+	}
+
+	cmd.Flags().String("behavior", "update", `Behavior when posting comment, one of:
+  update (default)  Update latest comment
+  new               Create a new comment
+  delete-and-new    Delete previous matching comments and create a new comment`)
+	_ = cmd.RegisterFlagCompletionFunc("behavior", func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		return validCommentBitbucketBehaviors, cobra.ShellCompDirectiveDefault
+	})
+	cmd.Flags().String("bitbucket-server-url", "", "Bitbucket Server URL")
```

Comment:
Does the URL default to something? Maybe that's defined elsewhere?

Reply:
Bitbucket Cloud is `bitbucket.org`, so I use it in the code to specify API endpoints (the hostname is hardcoded here: https://github.com/infracost/infracost/pull/1363/files#diff-b209df5ca9c5860317f0907a53d70e6889cbd10f3e352ae6bd8674becb38fe75R144). If this flag is specified though, it tells that users are using Bitbucket Server.

Reply:
Sorry what I meant is, why does the help text for github/lab show the default but this one doesn't? it would be nice if it could show the default but if that's not possible to do automatically, then changing the text to `Bitbucket Server URL, defaults to https://bitbucket.org` is enough. I think the text should include an example so the user knows if they should put https://, or any forward slashes at the end etc

```
      --github-api-url string   GitHub API URL (default "https://api.github.com")
      --gitlab-server-url string   GitLab Server URL (default "https://gitlab.com")
      --bitbucket-server-url string   Bitbucket Server URL
```

Reply:
Thanks!